src/v2i-hub/ImmediateForwardPlugin/src/ImmediateForwardConfiguration.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <optional>
+#include <unordered_map>
+#include <tmx/TmxException.hpp>
+#include <jsoncpp/json/json.h>
+
+
+namespace ImmediateForward
+{
+    static constexpr const char *NameKey = "name";
+    static constexpr const char *RSUSpecKey = "rsuSpec";
+    static constexpr const char *AddressKey = "address";
+    static constexpr const char *PortKey = "port";
+    static constexpr const char *MessagesKey = "messages";
+    static constexpr const char *TxModeKey = "txMode";
+    static constexpr const char *SignKey = "signMessages";
+    static constexpr const char *EnableHSMKey = "enableHsm";
+    static constexpr const char *HSMURLKey = "hsmUrl";
+    static constexpr const char *TmxTypeKey = "tmxType";
+    static constexpr const char *SendTypeKey = "sendType";
+    static constexpr const char *PSIDKey = "psid";
+    static constexpr const char *ChannelKey = "channel";
+
+
+    enum class TxMode
+    {
+        CONT = 0,
+        ALT = 1
+    };
+
+    enum class RSU_SPEC {
+        RSU_4_1 = 0,
+        NTCIP_1218 = 1
+    };
+    /**
+     * Message configuration
+     */
+    struct Message {
+        std::string tmxType;
+        std::string sendType;
+        std::string psid;
+        int channel;
+    };
+    /**
+     * Immediate forward configuration for an RSU connection
+     */
+    struct ImfConfiguration{
+
+        std::string name;
+        RSU_SPEC spec;
+        std::string address;
+        unsigned int port;
+        std::vector<Message> messages;
+        TxMode mode;
+        bool signMessage;
+        std::optional<bool> enableHsm;
+        std::optional<std::string> hsmUrl;
```

Comment:
Did we miss this hsmUrl parameter in the manifest.json? If we could add notes/description in doc or manifest.json to indicator which parameters are optional and mandatory, it would help with usability of this plugin.